pkg/apis/networking/v1alpha1/serverlessservice_lifecycle.go

```diff
@@ -46,6 +46,14 @@ func (sss *ServerlessServiceStatus) MarkEndpointsReady() {
 	serverlessServiceCondSet.Manage(sss).MarkTrue(ServerlessServiceConditionEndspointsPopulated)
 }
 
+// MarkDuplicateServices marks SKS as failed due to the fact that it owns
+// more than one private service.
+func (sss *ServerlessServiceStatus) MarkDuplicateServices() {
```

Comment:
This is never called?  I also think that you have a race condition we should discuss.  Grab me and we can chat.

Reply:
Removed.
Also added many private services code and tests